### PR TITLE
Upgrade canton to 20220209

### PR DESCRIPTION
**Pull request by oliverse-da**
Also:
- enabled deduplication tests
- removes pruning tests as canton no longer even pretends to support pruning (i.e. by now no longer pruning the ledger-api index either)

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [X] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by cocreature** [comment]
this conflicts with #12852

**Comment by oliverse-da** [comment]
> 

Thanks will back off and make test changes later.

**Comment by cocreature** [comment]
The other PR is merged so feel free to rebase & reopen.

File: ledger/ledger-api-test-tool-on-canton/BUILD.bazel [hunk 1]
```diff
@@ -81,12 +81,7 @@ conformance_test(
             "ParticipantPruningIT",  # see "conformance-test-participant-pruning" below
             "ClosedWorldIT",  # Canton currently fails this test with a different error (missing namespace in "unallocated" party id)
             "CommandServiceIT:CSsubmitAndWaitCompletionOffset",  # Canton does not fill the offsets
```
Comment (andreaslochbihler-da) [inline ledger/ledger-api-test-tool-on-canton/BUILD.bazel:83]: I can't find this test being excluded in Canton's own tests. So maybe this exclusion is now also obsolete?

Comment (andreaslochbihler-da) [reply ledger/ledger-api-test-tool-on-canton/BUILD.bazel:83]: Similarly for `PartyManagementServiceIT:PMAllocateEmptyExpectMissingDisplayName` below.

**Comment by oliverse-da** [comment]
Reopening PR with test-adaptations

**Pull request closed without merging**